### ISSUE TITLE
Questions about the Evaluation: Invalid Predicted Proofs

### ISSUE THREAD

**Issue by yangky11** [allenai/entailment_bank#7]
Hi,

I'm wondering how the evaluation should handle invalid predicted proofs. The current evaluation code doesn't seem to have a systematic way of handling it. For example, I was running the code with the ground truth proof:
`sent13 & sent19 -> int1: a new moon will occur 28 days after the last time it occurs; int1 & sent20 -> int2: the next new moon will occur 28 days after june 2; int2 & sent22 -> hypothesis;`
and a predicted proof produced by the pretrained T5-11B model:
`sent2 + 28 days = june 30; int1: the next new moon will occur 28 days after june 2; int2 & sent19 -> hypothesis;`

It produces an F1 score of 0.4 for the leaves, which is calculated from the following leaves:
```
ground truth leaves: sent13, sent19, sent20, sent22
predicted leaves: sent19
```

I'm wondering if the result makes sense. The predicted proof is not a valid tree after all. I think 0 would be better than 0.4 in this case.

An easy fix would be sampling multiple predictions from the model and filtering the invalid proofs. Was that used for producing the numbers in the paper?

Thanks!

**Comment by bhavanadalvi** [comment]
You are right, our evaluation code currently assumes the predicted proofs follow the proper syntax. So in this case, it has parsed the predicted proof as much as possible, to compute P=1.0, R=0.25, resulting in F1=0.4
Let me know what kind of checks are you looking at in your updated script to avoid such cases.

Re: _An easy fix would be sampling multiple predictions from the model and filtering the invalid proofs. Was that used for producing the numbers in the paper?_
That is an excellent suggestion, for the experiments in this paper we only looked at greedy best generations by the T5 models. But in some followup experiments we have observed that sampling multiple predictions can be helpful.
